### PR TITLE
Added support for look back in audience definition.

### LINKED ISSUE #21: Add support for lookback from audience definition.
## Summary

Audience definitions have a `lookback` field, which stipulates how far back in a user's local data we can check to find matching page views. Currently `engine` does not use this field when doing its evaluation and consumes the user's full data.

### Requirements
- Each audience definition can have its own `lookback` window which is a values in seconds, we must consider only the values which fit within this window.
- We should support a value which tells engine to use the full local data (`0`, `-1`, `Infinity`, ... ? )
- An entry in the `/docs` folder, we should document these features as part of the issue itself.

### PR DESCRIPTION
**Pull request by k11kirky**
closes #21 

Added documentation, implementation and tests for look back within audiences

### CODE REVIEW

File: docs/audiences.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+         automotiveInterestAudience } from '@airgrid/edgekit';
+```
+
+The following audiences 
+
+* sportInterestAudience
+* travelInterestAudience
+* automotiveInterestAudience
+
+for a full list of keywords see ```src/audiences/interest/.../keywords.ts```
```
Comment (ydennisy) [inline docs/audiences.md:63]: This list will soon be larger (before next release). There will be 30 audiences, you can see them in #3

File: test/run.test.ts [hunk 2]
```diff
@@ -18,13 +18,20 @@ const travelPageFeatureGetter = {
   },
 };
 
+const lookBackPageFeatureGetter = {
+  name: 'keywords',
+  func: (): Promise<string[]> => {
+    return Promise.resolve(['lb']);
```
Comment (ydennisy) [inline test/run.test.ts:24]: I think this is a little confusing to someone else, "why do we need 'lb' keyword here"

Might be good idea to match all page views when keywords are empty?

WDYT @Risto-Stevcev @k11kirky

File: test/run.test.ts [hunk 3]
```diff
@@ -33,27 +40,59 @@ const travelAudience: AudienceDefinition = {
...
 );
+
 const TWO_SPORTS_PAGE_VIEW_AFTER_TTL: Array<PageView> = pageViewCreator(
   timeStampInSecs() - TTL,
   ['sport'],
   2
 );
 
+const LOOK_BACK_INFINITY_PAGE_VIEW: Array<PageView> = pageViewCreator(
+  0,
```
Comment (ydennisy) [inline test/run.test.ts:85]: Are we setting this on purpose to show that anything will match?

Comment (k11kirky) [reply test/run.test.ts:85]: yes, anything in the past

File: test/run.test.ts (hunk 2 above)
Comment (k11kirky) [reply test/run.test.ts:24]: ok, thats fine to me

File: docs/audiences.md (hunk 1 above)
Comment (k11kirky) [reply docs/audiences.md:63]: @ydennisy Would you prefer not to list them individually? maybe we could point to the file that exports them all?

**Comment by k11kirky** [comment]
> I feel these tests are becoming quite hard to read, the audience definition and local data for each is setup at the top of the file, but the time I am at the bottom - I have no idea what the various variables contain.
> 
> Is there a better structure, which will read / look more like docs?

I was hoping the names would be self explanatory & we wouldn't need to reference back to them past the initial PR - hopefully we will not have to revisit the test too often and they should just tell us if the sdk is working or if we've made a breaking change. Open to other suggestions, we do reuse the audience definitions & localStoarge stuff, so It does need to be at the top of imported from a const file

**Pull request merged**